core/incremental_analysis.go

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright 2021-2024 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package core
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/JetBrains/qodana-cli/v2025/core/corescan"
+	"github.com/JetBrains/qodana-cli/v2025/platform"
+	"github.com/JetBrains/qodana-cli/v2025/platform/effectiveconfig"
+	"github.com/JetBrains/qodana-cli/v2025/platform/git"
+	"github.com/JetBrains/qodana-cli/v2025/platform/qdyaml"
+	"github.com/JetBrains/qodana-cli/v2025/platform/utils"
+	log "github.com/sirupsen/logrus"
+	"os"
+	"path/filepath"
+)
+
+var defaultRunner = &DefaultAnalysisRunner{}
+
+// AnalysisRunner defines the interface for logic on running analysis on commits
+type AnalysisRunner interface {
+	RunFunc(hash string, ctx context.Context, c corescan.Context) (bool, int)
+}
+
+// DefaultAnalysisRunner is the production implementation of RunFunc
+type DefaultAnalysisRunner struct {
```

Comment:
That's i believe bad name, cause default makes stron association with Default script.